pilot/pkg/proxy/envoy/v2/lds_test.go

```diff
@@ -416,9 +416,9 @@ func TestLDSWithSidecarForWorkloadWithoutService(t *testing.T) {
 		return
 	}
 
-	// Expect 1 HTTP listeners for 8081
-	if len(adsResponse.GetHTTPListeners()) != 1 {
-		t.Fatalf("Expected 1 http listeners, got %d", len(adsResponse.GetHTTPListeners()))
+	// Expect 3 HTTP listeners for 8081, 9080 and one virtualInbound
```

Comment:
this doesn't seem right, why did virtualInbound not have one before?

Can we actually check that the listeners are 8081, 9080, and virtualInbound instead of pretending to test it by just checking the length?

Reply:
It didn't happen before because the listener code would reject any ingressListener that did not match with a service instance. So you only got outbound listeners. With this PR, you start getting inbound listeners that correspond to ingressListener specs. So,  you get the 9080 one. In addition, due to the filter chain aggregation, the virtual Inbound is also categorized as a http listener because the ads client code marks any listener with any filter chain containing a HCM as a http listener :(.